presentation/components/src/presentation-components/instance-filter-builder/InstanceFilterConverter.ts

```diff
@@ -0,0 +1,201 @@
+  relatedInstances: RelatedInstanceDescription[];
+  propertyClasses: ClassInfo[];
+}
+
+function convertFilter(filter: PresentationInstanceFilter, ctx: ConvertContext) {
+  if (isFilterConditionGroup(filter))
+    return convertConditionGroup(filter, ctx);
+  return convertCondition(filter, ctx);
+}
+
+function convertConditionGroup(group: PresentationInstanceFilterConditionGroup, ctx: ConvertContext): string {
+  const convertedConditions = group.conditions.map((condition) => convertFilter(condition, ctx));
+  return `(${convertedConditions.join(` ${getGroupOperatorString(group.operator)} `)})`;
+}
+
+function convertCondition(condition: PresentationInstanceFilterCondition, ctx: ConvertContext): string {
+  const { field, operator, value } = condition;
+  const property = field.properties[0].property;
+  const relatedInstance = getRelatedInstanceDescription(field, property.classInfo.name, ctx);
+  addClassInfoToContext(relatedInstance ? relatedInstance.path[0].sourceClassInfo : property.classInfo, ctx);
+  const propertyAlias = relatedInstance?.alias ?? "this";
+
+  return createComparison(property, propertyAlias, operator, value);
+}
+
+function addClassInfoToContext(classInfo: ClassInfo, ctx: ConvertContext) {
+  if (ctx.propertyClasses.find((existing) => existing.id === classInfo.id))
+    return;
+
+  ctx.propertyClasses.push(classInfo);
+}
+
+function getRelatedInstanceDescription(field: PropertiesField, propClassName: string, ctx: ConvertContext): RelatedInstanceDescription | undefined {
+  if (!field.parent)
+    return undefined;
+
+  const pathToProperty = RelationshipPath.reverse(getPathToPrimaryClass(field.parent));
+  const existing = ctx.relatedInstances.find((instance) => RelationshipPath.equals(pathToProperty, instance.path));
+  if (existing)
+    return existing;
+
+  const newRelated = {
+    path: pathToProperty,
+    alias: `rel_${propClassName.split(":")[1]}`,
+  };
+
+  ctx.relatedInstances.push(newRelated);
+  return newRelated;
+}
+
+function getPathToPrimaryClass(field: NestedContentField): RelationshipPath {
+  if (field.parent) {
+    return [...field.pathToPrimaryClass, ...getPathToPrimaryClass(field.parent)];
+  }
+  return [...field.pathToPrimaryClass];
+}
+
+function createComparison(property: PropertyInfo, alias: string, operator: PropertyFilterRuleOperator, propValue?: PrimitiveValue): string {
+  const propertyAccessor = `${alias}.${property.name}`;
+  const operatorExpression = getRuleOperatorString(operator);
+  if (propValue === undefined || isUnaryPropertyFilterOperator(operator)) {
+    return `${propertyAccessor} ${operatorExpression}`;
+  }
+
+  const value = propValue.value;
+  if (operator === PropertyFilterRuleOperator.Like && typeof value === "string") {
+    return `${propertyAccessor} ${operatorExpression} "%${value.replace(/"/g, `""`)}%"`;
+  }
+
+  let valueExpression = "";
+  switch (typeof value) {
+    case "string":
+      valueExpression = `"${value.replace(/"/g, `""`)}"`;
```

Comment:
it seems we could extract this part into some `escapeStringValue` function to avoid repeated code